tutorial/mpi4py/src/mpi4py_reduce.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python
+"""
+MPI_Reduce on MPI rank. This should result in a total of (size * (size - 1) / 2),
+where size is the total number of ranks.
+Prints the total number of ranks, the sum of all ranks, and the time elapsed for the reduction."
+"""
+
+import argparse
+import time
+
+from mpi4py import MPI
+
+parser = argparse.ArgumentParser(description='mpi4py reduction benchmark',
+                                 formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+parser.add_argument('--n_warmup', type=int, default=100,
+                    help='Number of warmup iterations')
+parser.add_argument('--n_iter', type=int, default=1000,
+                    help='Number of benchmark iterations')
+args = parser.parse_args()
+
+n_warmup = args.n_warmup
+n_iter = args.n_iter
+
+size = MPI.COMM_WORLD.Get_size()
+rank = MPI.COMM_WORLD.Get_rank()
+name = MPI.Get_processor_name()
+
+# Warmup
+t0 = time.time()
+for i in range(n_warmup):
+    total = MPI.COMM_WORLD.reduce(rank)
+
+# Actual reduction, multiple iterations for accuracy of timing
+t1 = time.time()
+for i in range(n_iter):
+    total = MPI.COMM_WORLD.reduce(rank)
```

Comment:
i know SUM is the default, but maybe it's more instructive to set it explicitly?
```suggestion
    total = MPI.COMM_WORLD.reduce(rank, op=MPI.SUM)
```